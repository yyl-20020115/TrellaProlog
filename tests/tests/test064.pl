:- dynamic(p/2).
:- dynamic(p/3).

p(Z, Z).
clouds(are, nice).
p(Z, h(Z, W), f(W)).

main :-
	findall(Z, p(Z, Z), [Z]), write('ok1\n'),
	var(Z),
	findall(Z, p(Z, z), [z]), write('ok2\n'),
	findall(Z, p(Z, w), [w]), write('ok3\n'),
	\+ p(z, w), write('ok4\n'),
	p(w, w), write('ok5\n'),
	\+ clouds(Z, Z), write('ok6\n'),
	findall(Z, clouds(are, Z), [nice]), write('ok7\n'),
	\+ p(z, h(z, z), f(w)), write('ok8\n'),
	p(z, h(z, w), f(w)), write('ok9\n'),
	findall(W, p(z, h(z, W), f(w)), [w]), write('ok10\n'),
	findall(Z, p(Z, h(Z, w), f(Z)), [w]), write('ok11\n'),
	\+ p(z, h(Z, w), f(Z)), write('ok12\n'),
	retract(p(_,_,_)), write('ok13\n'),
	assertz(p(Z, h(Z, W), f(W))), write('ok14\n'),
	p(f(f(a)), h(f(f(a)), f(a)), f(f(a))), write('ok15\n'),
	retract(p(Z, h(Z, W), f(W))), write('ok16\n').

:- initialization(main).
