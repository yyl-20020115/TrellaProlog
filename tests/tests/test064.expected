ok1
ok2
ok3
ok4
ok5
ok6
ok7
ok8
ok9
ok10
ok11
ok12
ok13
ok14
ok15
ok16
